carbon-design-system/carbon
# fix: fixed focus flow issue on right arrow press

Closes #17281

The bug mentioned in the above issue, shows that the expected behaviour of right arrow press on a focussed and a opened node should bring down to the first child of the opened node is missing. This has been fixed in this PR.

The reference to the behaviour is on our website's `TreeView` component [page](https://carbondesignsystem.com/components/tree-view/usage/#behaviors:~:text=the%20node%20container.-,Keyboard%20controls,-Key)

#### Changelog

**New**

- renamed `ref` to `forwardRef` to distinguish between our local ref
- new `setRefs` function, which checks for `forwardRef` to say its a parent node or not
- passed the `setRefs` to the `li` in the jsx

**Changed**

- added some error handling to make sure that we don't have any console errors

#### Testing / Reviewing

- open the deploy preview of the `TreeView`
- in the default story, focus on the first item
- navigate to a closed node which has more leaf nodes in it
- press right arrow once, to see that the node opens
- press right arrow once again, and the focus should move to the first child of the opened parent node.
